File: app/src/main/java/com/appchamp/wordchunks/models/Pack.java
package com.appchamp.wordchunks.models;

import io.realm.RealmList;
import io.realm.RealmObject;


public class Pack extends RealmObject {

    private long id;

    private String name;

    private RealmList<Level> levels = null;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public RealmList<Level> getLevels() {
        return levels;
    }

    public void setLevels(RealmList<Level> levels) {
        this.levels = levels;
    }

}

File: app/src/main/java/com/appchamp/wordchunks/realmdb/models/realm/Pack.kt
package com.appchamp.wordchunks.realmdb.models.realm

import io.realm.RealmList
import io.realm.RealmModel
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass
import io.realm.annotations.Required


@RealmClass
open class Pack : RealmModel {

    @PrimaryKey
    @Required
    var id: String = ""
    @Required
    var title: String = ""
    @Required
    var color: String = ""
    var state: Int = 0  // LOCKED = 0, IN_PROGRESS = 1, FINISHED = 2
    var levels: RealmList<Level> = RealmList()

}


// States of pack
enum class PackState(val value: Int) {
    LOCKED(0),
    IN_PROGRESS(1),
    FINISHED(2)
}